src/stream.js

```diff
@@ -429,13 +429,14 @@ const scanCode = (opts, scope) => {
     throw new Error('Please specify \'containerId\' where the video element can be added as a child');
   }
 
-  // Begin the stream by selecting a read facing camera, the last usually being the HQ sensor
+  // Begin the stream by selecting a read facing camera, the last usually being the HQ sensor on
+  // Android devices
   return navigator.mediaDevices.enumerateDevices()
     .then((devices) => devices.filter((device) => device.kind === 'videoinput'))
     .then((devices) => navigator.mediaDevices.getUserMedia({
       video: {
         facingMode: 'environment',
-        deviceId: devices.length > 0 ? devices[devices.length - 1].deviceId : undefined,
+        deviceId: navigator.userAgent?.includes('Android') && devices.length > 0 ? devices[devices.length - 1].deviceId : undefined,
```

Comment:
This is getting complex - I'd make it a named variable. 

So this selection logic was doing bad things on iOS?